httpcache_test.go

```diff
@@ -1520,3 +1517,73 @@ func TestClientTimeout(t *testing.T) {
 		t.Error("client.Do took 2+ seconds, want < 2 seconds")
 	}
 }
+
+// TestTransport_varyMatches tests that the VaryIgnoreMask values are properly masked when
+// determining whether a cached response is returned or not.
+func TestTransportVaryMatchesWithMask(t *testing.T) {
+	resetTest()
+	s.transport.VaryIgnoreMask = map[string]bool{"Accept": true}
+	req, err := http.NewRequest("GET", s.server.URL+"/varyaccept", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Initial Accept header
+	req.Header.Set("Accept", "text/plain")
+	{
```

Comment:
why the blocks?

Reply:
to provide block scoping. other tests were doing it so I kept it for consistency.